api/basic/add-pin-router.ts

```diff
@@ -0,0 +1,250 @@
+import querystring from 'node:querystring'
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb'
+import { DynamoDBDocumentClient, GetCommand } from '@aws-sdk/lib-dynamodb'
+import { APIGatewayProxyEventV2 } from 'aws-lambda'
+import { CID } from 'multiformats/cid'
+import { Multiaddr } from 'multiaddr'
+import fetch from 'node-fetch'
+
+import { ClusterAddResponse, ClusterStatusResponse, PeerMapValue, Pin, Response } from './schema.js'
+import { doAuth } from './helper/auth-basic.js'
+
+interface AddPinInput {
+  cid: string
+  origins: string[]
+  dynamo: DynamoDBClient
+  table: string
+  indexerEndpoint: string
+  pickupEndpoint: string
+  token: string
+  balancerRate: number
+}
+
+/**
+ * AWS API Gateway handler for POST /pin/${cid}?&origins=${multiaddr},${multiaddr}
+ * Collect the params and delegate to addPin to do the work
+ *
+ * We provide responses in Payload format v2.0
+ * see: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format
+ */
+export async function handler (event: APIGatewayProxyEventV2): Promise<Response> {
+  const {
+    TABLE_NAME: table = '',
+    CLUSTER_BASIC_AUTH_TOKEN: token = '',
+    // set for testing
+    DYNAMO_DB_ENDPOINT: dbEndpoint = undefined,
+    INDEXER_ENDPOINT: indexerEndpoint = '',
```

Comment:
Misconception: There are no indexers involved in this process. What you're calling indexer is actually the IPFS Cluster. See my other comment on CLUSTER_IPFS_ADDR